presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java

```diff
@@ -19,33 +19,51 @@
 import com.facebook.presto.sql.parser.SqlParser;
 import com.facebook.presto.sql.planner.Symbol;
 import com.facebook.presto.sql.planner.plan.PlanNode;
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.Multimap;
 
-import java.util.List;
 import java.util.Map;
 
 /**
  * It is going to be executed at the end of logical planner, to verify its correctness
  */
 public final class PlanSanityChecker
 {
-    private static final List<Checker> CHECKERS = ImmutableList.of(
-            new ValidateDependenciesChecker(),
-            new TypeValidator(),
-            new NoSubqueryExpressionLeftChecker(),
-            new NoApplyNodeLeftChecker(),
-            new VerifyNoFilteredAggregations(),
-            new VerifyOnlyOneOutputNode());
+    private static final Multimap<Stage, Checker> CHECKERS = ImmutableListMultimap.<Stage, Checker>builder()
+            .putAll(
+                    Stage.INTERMEDIATE,
+                    new ValidateDependenciesChecker(),
+                    new TypeValidator(),
+                    new NoSubqueryExpressionLeftChecker())
+            .putAll(
+                    Stage.FINAL,
+                    new NoApplyNodeLeftChecker(),
+                    new VerifyNoFilteredAggregations(),
+                    new VerifyOnlyOneOutputNode())
+            .build();
 
     private PlanSanityChecker() {}
 
     public static void validate(PlanNode planNode, Session session, Metadata metadata, SqlParser sqlParser, Map<Symbol, Type> types)
     {
-        CHECKERS.forEach(checker -> checker.validate(planNode, session, metadata, sqlParser, types));
+        CHECKERS.values().forEach(checker -> checker.validate(planNode, session, metadata, sqlParser, types));
```

Comment:
Currently, yes we want to all validators. Also there are no duplicates so far, however having duplicates  does not matter much.